### DISCUSSION TITLE
Get and passing data to mopup

### DISCUSSION THREAD

**Discussion by jbdioli** [LuckyDucko/Mopups#121] [General]
Hello Guys

Sorry for my lack of knowledge. I would like to pass an object and retrieve an object from mopup and I don't know how to do it. 

I'm trying to understand your documentation, but unfortunately, I can't put it into practice :(

Could you help me? 

Thanks a lot

**Comment by LuckyDucko** [comment]
Ah yeah, so for a simple implementation, look at the demo project. It should have examples of returning things along with specific pages.

The skinny of it is that you will need to implement/inherit popupviewmodel<your_object_here>
And run with that. Call the inbuilt returns and that should be peachy.

Thats very opinionated in how i think things should work however, some people prefer a simpler implementation, using publisher subscribers for example. 

If you need more help, provide a more detailed vibe of the object you are trying to return, ill see what i can work with. I dont do Maui development anymore, but always happy to drop in for simple tasks
